harness/determined/cli/user.py

```diff
@@ -76,7 +76,11 @@ def log_in_user(args: Namespace) -> None:
     password = getpass.getpass(message)
 
     token_store = authentication.TokenStore(args.master)
-    utp = authentication.login(args.master, username, password, cli.cert)
+    try:
```

Comment:
I'd prefer to see this tested in a unit test (https://github.com/determined-ai/determined/blob/main/harness/tests/cli/test_user.py) than in an end-to-end test. I think it's ok to assume that `authentication.login` throws an `UnauthenticatedException` when trying to log in with a bad username, and to mock that behavior in a test.

Actually relying on a master to send a "yup, can't login" is heavier-weight than we need, and end-to-end tests are generally harder to debug.

https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1351909582/Python+Testing might be a helpful resource, and please also write me on Slack if you'd like help with this part.